site/source/tutorials/working_with_virtual_dom.md

```diff
@@ -0,0 +1,106 @@
+
+The DOM API has existed for many years and has been continually expanded to keep up with the evolution of web browsers. This evolution has yielded a programming interface that is occasionally verbose and inconsistent. Virtual DOM libraries are able to reconsider the use-cases for managing page elements and expose a simpler set of interfaces for developers to work with. For example, consider the example in the preceeding section where we created an `h1` tag, set some properties on it, and add it to the page. In Dojo 2, this would be accomplished using the following code:
+
+```ts
+  v('h1', { class: 'title' }, [ 'Welcome to Biz-E-Corp' ]);
+```
+
+A single line of code is all that is required to accomplish what the DOM API required four to do. By providing a simple, declarative API, rendering code is typically much cleaner and easier to understand.
+
+One critical difference between the DOM and virtual DOM examples is that the DOM example maintains access to the element that was created via the `pageTitle` variable. Keeping a reference like this is often done when the element needs to be changed later. With a virtual DOM, maintaining state in this way is not necessary because the application never manipulates the state of the DOM directly. It is only responsible for generating a representation of the page at any given time. A component of the virtual DOM called a [projector](http://maquettejs.org/docs/typedoc/interfaces/_maquette_.projector.html) is responsible to determine the changes that are required to the page.
+
+{% aside 'Virtual DOM projectors' %}
+A virtual DOM renders content to the web page using the DOM just like any other web application. A **projector** is a component of the virtual DOM that synchronizes the virtual DOM nodes and the DOM.
+{% endaside %}
+
+Consider an example where a reuseable page title component is required. This widget must be able to vary the content inside of an `h1` element. Using the DOM, we might end up with a method like this:
+
+```ts
+  updateTitle(title: string): void {
+    this.pageTitle.innerHTML = title;
+  }
+```
+
+This code requires the widget instance to track the DOM element that contains the title text and set the correct property whenever it is called. In contrast, the same thing is done in Dojo 2 with this method:
+
+```ts
+  render(): DNode {
+    return v('h1', { class: 'title' }, [ this.title ]);
+  }
+```
+
+Notice how similar this is to the original statement that returned static content? Since the projector automatically determines the differences, the application does not have to track changes - it simply describes the elements as they should exist. This greatly reduces the API required to describe the content of a web page.
```

Comment:
Notice how similar this is to the original statement that returned static content? -> Do you notice the similarity between this and the original statement that returned static content?